test/service/test_pagination.py

```diff
@@ -34,23 +35,33 @@ def assert_page1_correct(self, json_response):
         :param json_response: A JSON response dictionary
         :return:
         """
-        if 'search_before' in json_response['pagination']:
-            self.assertIsNone(json_response['pagination']['search_before'],
-                              "search_before != null on first page of results")
-            self.assertIsNone(json_response['pagination']['search_before_uid'],
-                              "search_before != null on first page of results")
+        if 'previous' in json_response['pagination']:
+            self.assertIsNone(json_response['pagination']['previous'],
+                              "previous != null on first page of results")
         else:
             self.fail("search_before not set to null on first page of results")
 
         num_hits = len(json_response['hits'])
-        if 'search_after' in json_response['pagination']:
-            self.assertEqual(json_response['pagination']['search_after'],
+        if 'next' in json_response['pagination']:
+            search_after, search_after_uid = self._get_paging_params_from_url(json_response, 'next')
+            self.assertEqual(search_after,
                              json_response['hits'][num_hits - 1]['entryId'],
-                             "search_after not set to last returned document on first page")
-            self.assertIsNotNone(json_response['pagination']['search_after_uid'])
+                             "next not set to last returned document on first page")
+            self.assertIsNotNone(search_after_uid)
         else:
             self.fail("search_after not set on first page of results")
 
+    def _get_paging_params_from_url(self, json_response, string):
+        query = urlparse(json_response['pagination'][string]).query
+        try:
+            _, search, search_uid = query.split('&')
```

Comment:
Instead of parsing the query string yourself which could break if the ordering of query string params is ever different, have you considered using `parse_qs`?
```
parsed_query = parse_qs(query)
search = parsed_query.get('search_after', [None])[0]
search_uid = parsed_query.get('search_after_uid', [None])[0]
```

Reply:
It would make the duality of the function (working for both search_after and search_before) a little more tricky though